src/bindings/PythonBindings.cpp

```diff
@@ -102,7 +102,8 @@ PYBIND11_MODULE(CesiumOmniversePythonBindings, m) {
         .def_readonly("percent_complete", &CesiumIonClient::Asset::percentComplete);
 
     py::class_<CesiumIonClient::Connection>(m, "Connection")
-        .def("get_api_url", &CesiumIonClient::Connection::getApiUrl)
+        // Wrap non-static member function in lambda. May be able to use py::overload_cast<> in C++ 20
+        .def("get_api_uri", [](CesiumIonClient::Connection& connection) { return connection.getApiUrl(); })
```

Comment:
I couldn't figure out how to do this in a cleaner way. `getApiUrl` has two overloads:

```c++
  static CesiumAsync::Future<std::optional<std::string>> getApiUrl(
      const CesiumAsync::AsyncSystem& asyncSystem,
      const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
      const std::string& ionUrl);
```

```c++
  const std::string& getApiUrl() const noexcept { return this->_apiUrl; }
```

First I tried `py::overload_cast<>`, which didn't work because it expects at least one template argument
Then I tried the `static_cast` approach in https://pybind11.readthedocs.io/en/stable/classes.html
I ended up just wrapping it in a lambda.

@weegeekps am I missing an obvious way to do this?

Reply:
Seems like dynamic return types are only supported using a pattern similar to what you're already doing. I think the open question is whether or not you need to wrap `connection.getApiUrl()` in a `py::cast` call.